.github/workflows/acceptance-tests-runner.yml

```diff
@@ -1269,8 +1269,13 @@ jobs:
         run: make testacc
 
   resource_policy:
-    needs: [ change-detection, get-provider-version ]
-    if: ${{ needs.change-detection.outputs.resource_policy == 'true' || inputs.test_group == 'resource_policy' }}
+    # Resource policy tests create org-level policies that can block AWS cluster creation.
+    # Run after advanced_cluster to avoid interfering with cluster tests, or immediately if advanced_cluster is skipped.
+    needs: [ change-detection, get-provider-version, advanced_cluster ]
```

Comment:
nit: There are other test group which also create cluster as part of testing (e.g. `search_index`), is there a chance it can impact those creations as well?